tools/build_api.py

```diff
@@ -230,11 +231,15 @@ def find_valid_toolchain(target, toolchain):
     """
     end_warnings = []
     toolchain_names = get_valid_toolchain_names(target, toolchain)
+    if(target.default_toolchain == "uARM"):
+        end_warnings.append(UARM_TOOLCHAIN_WARNING)
     last_error = None
     for index, toolchain_name in enumerate(toolchain_names):
         internal_tc_name = get_toolchain_name(target, toolchain_name)
         if toolchain == "ARM" and toolchain_name == "ARMC5" and index != 0:
             end_warnings.append(ARMC5_MIGRATION_WARNING)
+        if toolchain_name == "uARM":
```

Comment:
Is it not necessary to also check `toolchain == "ARM"` as done in the previous `if`?

You could have:

```suggestion
        if toolchain == "ARM":
            if toolchain_name == "ARMC5" and index != 0:
                end_warnings.append(ARMC5_MIGRATION_WARNING)
            else if toolchain_name == "uARM":
                end_warnings.append(UARM_TOOLCHAIN_WARNING)
```

Reply:
I am not very clear on how the target toolchain gets constructed so I thought if in case build mbed compile with "uARM" toolchain so appending deprecate message

Reply:
So you are completely ignoring `toolchain` and only consider `toolchain_name`?

Reply:
As I have mentioned I am not very clear,  how the toolchain gets constructed

Reply:
> As I have mentioned I am not very clear,  how the toolchain gets constructed but quickly will give try.

Reply:
We can use two ways to use uARM toolchain
1.uARM toolchain pass as an argument "mbed compile -t uARM"
2.Define default_toolchain as uARM in targets.json to the target device

for the first case we can check toolchain_name with uARM and append the deprecate message and for the second case if we try to use mbed compile -t ARMC6 and your target device default_toolchain is uARM then build tool internally change to ARMC6 with microlib(this happens later) so for the second case I am checking default_toolchain is uARM then appending the deprecate message

So we need both check to raise deprecate